lib/proxyquire.js

```diff
@@ -129,17 +130,47 @@ Proxyquire.prototype.load = function (request, stubs) {
   return this._withoutCache(this._parent, stubs, request, this._parent.require.bind(this._parent, request))
 }
 
+// Resolves a stub relative to a module.  Note that this only resolves
+// "relative" paths (e.g. "./foo") but not non-relative modules (e.g. "react").
+// Also, if noPreserveCache(), then we won't throw if the module can't
+// be resolved.
+Proxyquire.prototype._resolveRelativePath = function (path, moduleFilename) {
+  try {
+    return resolve.sync(path, {
+      basedir: dirname(moduleFilename),
+      extensions: Object.keys(require.extensions),
+      paths: Module.globalPaths
+    })
+  } catch (err) {
+    // If this is not a relative path (e.g. "foo" as opposed to "./foo"), and
+    // we couldn't resolve it, then we just let the path through unchanged.
+    if (path[0] !== '.') {
+      return path
+    }
+
+    // If !this._preserveCache, then we don't want to throw, since we can
+    // resolve modules that don't exist.  Resolve as best we can.
+    if (!this._preserveCache) {
+      return pathResolve(dirname(moduleFilename), path)
+    }
+
+    throw err
+  }
+}
+
 // This replaces a module's require function
 Proxyquire.prototype._require = function (module, stubs, path) {
   assert(typeof path === 'string', 'path must be a string')
   assert(path, 'missing path')
 
-  if (hasOwnProperty.call(stubs, path)) {
-    var stub = stubs[path]
+  var resolvedPath = this._resolveRelativePath(path, module.filename)
+
+  if (hasOwnProperty.call(stubs, resolvedPath)) {
+    var stub = stubs[resolvedPath]
 
     if (stub === null) {
       // Mimic the module-not-found exception thrown by node.js.
-      throw moduleNotFoundError(path)
+      throw moduleNotFoundError(resolvedPath)
```

Comment:
Shouldn't this be the original user input? This is simulating the error message node would generate and that would be based on the `require` id the user passes, not the resolved path.

Reply:
Yes.  :)